core-bundle/assets/controllers/jobs-controller.js

```diff
@@ -0,0 +1,67 @@
+import { Controller } from '@hotwired/stimulus';
+import { TurboStreamConnection } from '../modules/turbo-stream-connection';
+
+export default class extends Controller {
+    _turboStreamConnection = new TurboStreamConnection();
+    _runningJobs = false;
+
+    static values = {
+        pendingJobsUrl: String,
+        defaultInterval: Number,
+        maximumInterval: Number,
+        enabled: Boolean,
+    };
+
+    static targets = ['count', 'list'];
+
+    connect() {
+        this._pollInterval = this.defaultIntervalValue;
+        this._timer = null;
+
+        if (this.enabledValue) {
+            this.enable();
+        }
+    }
+
+    enable() {
+        clearTimeout(this._timer);
+        this._poll();
+    }
+
+    listTargetConnected(el) {
+        // Clear timer in case the target was added manually
+        clearTimeout(this._timer);
+
+        if (el.dataset.jobs === '0') {
+            this.countTarget.innerText = '';
+
+            if (this._runningJobs) {
+                this._runningJobs = false;
+                this._allJobsProcessed();
+            }
+
+            // Continuously increase interval if there are no results
+            this._pollInterval = Math.min(this.maximumIntervalValue, this._pollInterval * 2);
+        } else {
+            this.countTarget.innerText = el.dataset.jobs;
+            this._runningJobs = true;
+
+            this._pollInterval = this.defaultIntervalValue;
+        }
+
+        this._waitAndPoll();
+    }
+
+    _allJobsProcessed() {
+        // todo: replace with flash message once the functionality from #8204 has been merged
+        console.log('All jobs have been processed.');
```

Comment:
```suggestion
        // todo: add a flash message once the functionality from #8204 has been merged
```

As long as #8204 isn't merged, I think we can remove this

Reply:
Fine for me. But then let's maybe add a note to #8204 so that we don't forget about it.

Reply:
I would leave that as is, maybe we can get #8204 merged before, I just pinged people on Slack ;)

Reply:
Done in 3f4e28fe4093d9eb5affa5596aaa5738713ce765.